### DISCUSSION TITLE
Jersey client 2.21 doesnt support Transfer-encoding: chunked, is there any workaround?

### DISCUSSION THREAD

**Discussion by svatos-jirka** [dropwizard/dropwizard#7939] [Q&A]
HI,

we are using Jersey client 2.21, but customer changed the server to enable transfer-encoding to Chunked, which the client is not able to handle. Please, does exist some workaround how to solved on client side?

many thanks, Jiri

**Comment by zUniQueX** [comment]
Hi @svatos-jirka. Your description states you're using the Jersey client, but it doesn't say you're using Dropwizard. If you have a question related to Jersey only, please ask at their project. They may give you a better reply. Additionally, Jersey 2.21 is rather old and you may want to migrate to a newer version.

However, your description doesn't provide enough information to give you a proper workaround. The Jersey client is a high-level client API and defers the HTTP logic to `Connector`s, which can be registered with [`ConnectorProvider`s](https://javadoc.io/static/org.glassfish.jersey.core/jersey-client/2.21/org/glassfish/jersey/client/spi/ConnectorProvider.html). If you're not registering one, the default `HttpUrlConnectorProvider` is used. The encoding can be changed with the property `REQUEST_ENTITY_PROCESSING`, but some connectors only work well with one encoding. The [JavaDoc of the `HttpUrlConnectorProvider`](https://javadoc.io/static/org.glassfish.jersey.core/jersey-client/2.21/org/glassfish/jersey/client/HttpUrlConnectorProvider.html) states it'll not work well with chunked encoding, but also gives a workaround. If it doesn't work for you, you may want to take a look at the [other connectors](https://eclipse-ee4j.github.io/jersey.github.io/documentation/2.21/user-guide.html#d0e4807).
